flake.nix

```diff
@@ -4,63 +4,94 @@
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
     flake-utils.url = "github:numtide/flake-utils";
+    naersk.url = "github:nix-community/naersk";
     nixgl.url = "github:guibou/nixGL";
   };
 
-  outputs = { self, nixpkgs, flake-utils, nixgl }:
+  outputs = { self, nixpkgs, flake-utils, naersk, nixgl }:
     flake-utils.lib.eachDefaultSystem
       (system:
         let
           pkgs = import nixpkgs {
             inherit system;
             overlays = [ nixgl.overlay ];
           };
-
+          naersk' = pkgs.callPackage naersk { };
           nao_sdk_version = "5.9.0";
           nao_sdk_environment_path = "$HOME/.naosdk/${nao_sdk_version}/environment-setup-corei7-64-aldebaran-linux";
+          buildInputs = with pkgs;[
+            # Tools
+            cargo
+            rustc
+            rustfmt
+            cmake
+            pkg-config
+            llvmPackages.clang
+            python312
+            rsync
+
+            # Libs
+            luajit
+            systemdLibs
+            hdf5
+            alsa-lib
+            opusfile
+            libogg
+            libGL
+            libxkbcommon
+            wayland
+            xorg.libX11
+            xorg.libXcursor
+            xorg.libXi
+            xorg.libXrandr
+            pkgs.nixgl.auto.nixGLDefault
+            rustPlatform.bindgenHook
+            openssl
+          ];
+
+          mktool = manifest_path:
+            let
+              manifest = (pkgs.lib.importTOML manifest_path).package;
+            in
+            naersk'.buildPackage {
+              name = manifest.name;
+              version = manifest.version;
+              src = ./.;
+              inherit buildInputs;
+              cargoBuildOptions = x: x ++ [ "-p" manifest.name ];
+              cargoTestOptions = x: x ++ [ "-p" manifest.name ];
+            };
         in
         {
+          packages.twix = mktool ./tools/twix/Cargo.toml;
+          packages.pepsi = mktool ./tools/pepsi/Cargo.toml;
+          packages.fanta = mktool ./tools/fanta/Cargo.toml;
```

Comment:
We have some more tools which don't see much use, but the `behavior_simulator` is used quite regularly.
It also has some lua dependencies which can be a bit annoying. Perhaps managing them with nix might be easier?

Also, `annotato`, which probably also needs the nixGL treatment. 😉